Makefile

```diff
@@ -18,11 +18,9 @@ install: all
 	cp .version zsh-syntax-highlighting.zsh $(SHARE_DIR)
 	cp COPYING.md README.md changelog.md $(DOC_DIR)
 	sed -e '1s/ .*//' -e '/^\[build-status-[a-z]*\]: /d' < README.md > $(DOC_DIR)/README.md
-	if [ x"true" = x"`git rev-parse --is-inside-work-tree 2>/dev/null`" ]; then \
-		git rev-parse HEAD; \
-	else \
-		cat .revision-hash; \
-	fi > $(SHARE_DIR)/.revision-hash
+	if prefix=`git rev-parse --show-prefix 2>/dev/null` && [ x"$$prefix" = x ]; then \
```

Comment:
IIRC autoconf uses test(1) rather than [(1) for comparing strings, but as you said in another context, let's keep it as it is unless someone complains.